packages/block-library/src/image/edit.js

```diff
@@ -246,6 +247,16 @@ class ImageEdit extends Component {
 		};
 	}
 
+	getFilename( url ) {
+		if ( url ) {
+			const fileName = url.match( /.*\/(.+?)(?:\?.*)?$/ );
+			if ( fileName ) {
+				return fileName[ 1 ];
+			}
+		}
+		return '';
```

Comment:
Won't this produce a nonsense `alt` value of `'This image has an empty alt attribute; its file name is '` if this `return` is reached?

Reply:
@aduth Thanks! I do not believe there is an instance that `url` will not pass given the Media Library is where this is passed back from. I can't imagine how it would not return `url`. I only added this check as an edge case precaution of a graceful return instead of an error of using an undefined `url`. Can you think of any edge cases where `url` would return false? Happy to remove the check; I just want to make sure that I'm not taking away a safety net. :)

Reply:
Well, ideally we're certain the sort of data we're expecting. For something like URL user input, it's a bit harder to nail down. But maybe it's safe to assume that every URL should at least have a `/` followed by a "filename" (this is what the pattern is checking, yeah? The fact I'm asking is maybe a hint it's not very obvious on its own). If we are going to be uncertain, we should at least follow-through on our uncertainty, where `defaultedAlt` doesn't perform the substitution at all if the URL can't be determined.